src/it/java/servlets/admin/moduleManagement/OpenAllModulesIT.java

```diff
@@ -0,0 +1,166 @@
+    @Test
+    public void testWithUserAuth()
+    {
+        try
+        {
+            log.debug("Signing in as Normal User Through LoginServlet");
+            testUtils.TestProperties.loginDoPost(log, request, response, testUsers[1], testUsers[1], null, LANGUAGE_CODE);
+            log.debug("Login Servlet Complete, Getting CSRF Token");
+
+            String csrfToken = response.getCookie("token").getValue();
+            assertFalse(csrfToken.isEmpty());
+
+            //Add Cookies from Response to outgoing request
+            request.setCookies(response.getCookies());
+            String responseBody = doMockPost(csrfToken, "enable");
+
+            assertTrue(responseBody.contains("loggedOutSheep"));
+        }
+        catch (Exception e)
+        {
+            log.fatal("Could not Complete: " + e.toString());
+            fail("Could not Complete: " + e.toString());
+        }
+    }
+
+    @Test
+    public void testWithAdminAuthUnsafeLevelsClosed()
+    {
+        String unsafe = "disable";
+        try
+        {
+            log.debug("Signing in as Admin User Through LoginServlet");
+            TestProperties.loginDoPost(log, request, response, testUsers[0], testUsers[0], null, LANGUAGE_CODE);
+            log.debug("Login Servlet Complete, Getting CSRF Token");
+
+            String csrfToken = response.getCookie("token").getValue();
+            assertFalse(csrfToken.isEmpty());
+
+            //Add Cookies from Response to outgoing request
+            request.setCookies(response.getCookies());
+            String responseBody = doMockPost(csrfToken, unsafe);
+
+            log.debug("Response Body: " + responseBody);
+
+            assertTrue(responseBody.contains("All Modules are Now Open"));
+
+        }
+        catch (NullPointerException e) { fail("Null Pointer" + e.toString());}
+        catch (Exception e) { fail(e.toString()); }
+    }
+
+
+    @Test
+    public void testWithAdminAuthUnsafeLevelsOpen()
+    {
+        String unsafe = "enable";
+        try
+        {
+            log.debug("Signing in as Admin User Through LoginServlet");
+            TestProperties.loginDoPost(log, request, response, testUsers[0], testUsers[0], null, LANGUAGE_CODE);
+            log.debug("Login Servlet Complete, Getting CSRF Token");
+
+            String csrfToken = response.getCookie("token").getValue();
+            assertFalse(csrfToken.isEmpty());
+
+            //Add Cookies from Response to outgoing request
+            request.setCookies(response.getCookies());
+            String responseBody = doMockPost(csrfToken, unsafe);
+
+            log.debug("Response Body: " + responseBody);
+
+            assertTrue(responseBody.contains("[WARNING] Server is vulnerable. Unsafe levels open!"));
+            assertTrue(responseBody.contains("All Modules are Now Open"));
+
+        }
+        catch (NullPointerException e) { fail("Null Pointer" + e.toString());}
+        catch (Exception e) { fail(e.toString()); }
+    }
+
+}
```

Comment:
Perhaps an Unauthenticated test as well considering the danger of the function?

Reply:
Test has been added